app/components/work_package_relations_tab/relations_mediator.rb

```diff
@@ -29,21 +29,67 @@
 #++
 
 class WorkPackageRelationsTab::RelationsMediator
+  RelationGroup = Data.define(:type, :visible_relations, :ghost_relations)
+
   attr_reader :work_package
 
   def initialize(work_package:)
     @work_package = work_package
   end
 
-  def relations
-    @relations ||= work_package.relations.includes(:to, :from).visible
+  def visible_relations
+    @visible_relations ||= work_package.relations.includes(:to, :from).visible
+  end
+
+  def visible_children
+    @visible_children ||= work_package.children.visible
+  end
+
+  def ghost_relations
+    @ghost_relations = work_package.relations.includes(:to, :from).where.not(id: @visible_relations.select(:id))
```

Comment:
If `visible_relations` has not been called yet, then `@visible_relations` would be `nil`, and that method would not work properly.

```suggestion
    @ghost_relations = work_package.relations.includes(:to, :from).where.not(id: visible_relations.select(:id))
```

It should use `visible_relations` directly, so that it works even if `visible_relations` has never been called before.